oauth2/handler.go

```diff
@@ -1154,3 +1207,175 @@ func (h *Handler) logOrAudit(err error, r *http.Request) {
 		x.LogAudit(r, err, h.r.Logger())
 	}
 }
+
+// swagger:route POST /credentials oAuth2 createVerifiableCredential
+//
+// # Issues a Verifiable Credential
+//
+// This endpoint creates a verifiable credential that attests that the user
+// authenticated with the provided access token owns a certain public/private key
+// pair.
+//
+// More information can be found at
+// https://openid.net/specs/openid-connect-userinfo-vc-1_0.html.
+//
+//	Consumes:
+//	- application/json
+//
+//	Schemes: http, https
+//
+//	Responses:
+//	  200: verifiableCredentialResponse
+//	  400: verifiableCredentialPrimingResponse
+//	  default: errorOAuth2
+func (h *Handler) createVerifiableCredential(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	session := NewSessionWithCustomClaims("", h.c.AllowedTopLevelClaims(ctx))
+	accessToken := fosite.AccessTokenFromRequest(r)
+	tokenType, _, err := h.r.OAuth2Provider().IntrospectToken(ctx, accessToken, fosite.AccessToken, session)
+
+	if err != nil {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(err))
+		return
+	}
+	if tokenType != fosite.AccessToken {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(fosite.ErrInvalidRequest.WithHintf("the token is not an access token")))
```

Comment:
Hints are regular sentences and have proper punctuation:

`The provided token is not an access token.`